generators/spring-controller/prompts.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright 2013-2017 the original author or authors from the JHipster project.
+ *
+ * This file is part of the JHipster project, see http://www.jhipster.tech/
+ * for more information.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const jhiCore = require('jhipster-core');
+
+module.exports = {
+    askForControllerActions
+};
+
+function askForControllerActions() {
+    const done = this.async();
+
+    askForControllerAction.call(this, done);
+}
+
+function askForControllerAction(done) {
```

Comment:
You can move this method inside the `askForControllerActions` and convert this into a const arrow like below so that you dont have to bind `this`
```
function askForControllerActions() {
  const askForControllerAction = done => {
    ...
    this.prompt(prompts).then((props) => {
      if (props.actionAdd) {
        ...
        askForControllerAction(done);
      }
    });
  }
  const done = this.async();
  askForControllerAction(done);
}
```